### PR TITLE
added import and export scp resource

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
##### SUMMARY

Server Configuration Profile Import and Export Resource

##### ISSUE TYPE

- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below
/usr/local/go/bin/go test -timeout 1h -run TestAccRedfishSCP -coverprofile=coverage.out -cover terraform-provider-redfish/redfish/provider -v
```
image

#### Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

#### Output from acceptance testing:
image

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

#### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Review by Sakshi-dell — DISMISSED** [review]

File: redfish/provider/resource_redfish_scp_export.go [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+	}
+}
+
+// ShareParametersExportSchema returns the schema for the share parameters
+func ShareParametersExportSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"filename": schema.StringAttribute{
+			MarkdownDescription: "File Name - The name of the server configuration profile file to export. This is the name of the file " +
+				"that was previously exported using the Server Configuration Profile Export operation. This file is typically " +
+				"in the xml/json format",
```
Comment (rounak-adhikary) [inline redfish/provider/resource_redfish_scp_export.go:186]: Please check if the 2nd and last lines are relevant for export.

File: redfish/provider/resource_redfish_scp_export.go [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+func (*ScpExportResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	// Get Plan Data
+	var plan models.TFRedfishScpExport
+	diags := req.Config.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	sp := plan.ShareParameters
+	shareType := sp.ShareType.ValueString()
```
Comment (rounak-adhikary) [inline redfish/provider/resource_redfish_scp_export.go:400]: Shares are optional, so
```suggestion
	sp := plan.ShareParameters
	if sp.IsNull() || sp.IsUnknown() {
	        return
	}
	shareType := sp.ShareType.ValueString()
```

Comment (rounak-adhikary) [reply redfish/provider/resource_redfish_scp_export.go:400]: Same for import resource

Comment (Krishnan-Priyanshu) [reply redfish/provider/resource_redfish_scp_export.go:400]: Share parameter should be required. 
will change it to required

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**